library/alloc/src/str.rs

```diff
@@ -383,15 +383,45 @@ impl str {
                   without modifying the original"]
     #[stable(feature = "unicode_case_mapping", since = "1.2.0")]
     pub fn to_lowercase(&self) -> String {
-        let mut s = String::with_capacity(self.len());
-        for (i, c) in self[..].char_indices() {
+        let mut out = Vec::<u8>::with_capacity(self.len());
+        let b = self.as_bytes();
+
+        const UNROLL: usize = 128;
+        while out.len() + UNROLL <= b.len() {
+            let n = out.len() + UNROLL;
+            // Safety:
+            // we are working with ascii only at the moment
+            // so we know the exact size of the buffer
+            unsafe {
+                for j in out.len()..n {
+                    if !b.get_unchecked(j).is_ascii() {
+                        break;
+                    }
+                }
```

Comment:
This loop doesn’t do anything AFAICT. I think you meant for the `break` to refer to the outer loop instead.

Reply:
Oh shoot, yes you are correct